programs/util.c

```diff
@@ -167,7 +167,7 @@ int UTIL_utime(const char* filename, const stat_t *statbuf)
      * that struct stat has a struct timespec st_mtim member. We need this
      * check because there are some platforms that claim to be POSIX 2008
      * compliant but which do not have st_mtim... */
-#if (PLATFORM_POSIX_VERSION >= 200809L) && defined(st_mtime)
+#if (PLATFORM_POSIX_VERSION >= 200809L) && defined(st_mtime) || defined(__alpha__)
```

Comment:
So I guess it means `__alpha__` platform never claims POSIX 2008 compliance, since it probably is not,
yet nonetheless support `struct timespec` and `st_mtime` as extended features ?

The surprising part here is that, in this construction, `__alpha__` is presumed enough of a hint to _guarantee_ the support of `struct timespec`, no matter the software level. I would have expected `__alpha__` to be much older than `struct timespec`, and therefore logically, older variants of software for `__alpha__` would not provide `struct timespec`. Though I don't know enough about `__alpha__` history to be sure about that.

Another avenue could have been to declare `PLATFORM_POSIX_VERSION = 200809L` when the right set of conditions is present (`__alpha__` and maybe another condition), but this would imply support of other stuff beyond `struct timespec`.

Reply:
What about : 
```
#if (PLATFORM_POSIX_VERSION >= 200809L || defined(__alpha__)) && defined(st_mtime) 
```

Reply:
@Cyan4973, as the linked bug states, Alpha does not define `st_mtime` (which makes it non-compliant with POSIX). So that won't work.

Reply:
okay,
so `st_mtime` is in fact not required,
it's just employed as a way to improve odds that `struct timespec.st_mtim` is actually supported ?

Reply:
Correct, since we previously found some platform (I forget which) which claims to be POSIX 2008 but doesn't have `struct timespec` and we detect that by observing that they don't define `st_mtime`.